Production/govuk_ios/Deeplinking/DeeplinkDataStore.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+
+typealias URLPattern = String
+
+struct DeeplinkDataStore {
+    private let routes: [DeeplinkRoute]
+
+    init(routes: [DeeplinkRoute]) {
+        self.routes = routes
+    }
+
+    func route(for url: URL) -> ResolvedDeeplinkRoute? {
+        guard isValidDeeplink(url: url)
+        else { return nil }
+
+        return routes
+            .lazy
+            .map {
+                DeeplinkRoutingComponents(
+                    sourceURL: url,
+                    route: $0
+                )
+            }
+            .first { $0.hasMatchingUrls() }
+            .map { .init(components: $0) }
+    }
+
+    private func isValidDeeplink(url: URL) -> Bool {
+        !url.isFileURL &&
+        !url.pathComponents.isEmpty
+    }
+}
+
+extension DeeplinkDataStore {
```

Comment:
Why not in the extensions dir? https://github.com/alphagov/govuk-mobile-ios-app/tree/a9ff701c8b118d2a06c2ab6ea9a927f824cd6a33/Production/govuk_ios/Extensions